### PR TITLE
optimize: improve hlog example

### PR DESCRIPTION
**Pull request by L2ncE**
#### What type of PR is this?

optimize

#### What this PR does / why we need it (English/Chinese):

Changed hertz's logger extension to [hertz-contrib/logger](https://github.com/hertz-contrib/logger), and add log deletion and compression to the logrus example.

#### Which issue(s) this PR fixes:

None.

### CODE REVIEW

**Comment by li-jin-gou** [comment]
you can use `go mod tidy && go mod verify`

**Comment by L2ncE** [comment]
> you can use `go mod tidy && go mod verify`

done

File: go.mod [hunk 1]
```diff
@@ -30,18 +31,18 @@ require (
...
 	go.opentelemetry.io/proto/otlp v0.18.0 // indirect
-	go.uber.org/atomic v1.10.0 // indirect
-	go.uber.org/zap v1.23.0 // indirect
+	go.uber.org/zap v1.23.0
 	golang.org/x/net v0.0.0-20220630215102-69896b714898 // indirect
 	golang.org/x/sync v0.0.0-20220907140024-f12130a52804 // indirect
 	google.golang.org/genproto v0.0.0-20220630174209-ad1d48641aa7 // indirect
 	google.golang.org/protobuf v1.28.1
 	gopkg.in/ini.v1 v1.67.0 // indirect
+	gopkg.in/natefinch/lumberjack.v2 v2.0.0
```
Comment (li-jin-gou) [inline go.mod:47]: 添加 licenses

Comment (L2ncE) [reply go.mod:47]: > add licenses

done

File: hlog/logrus/main.go [hunk 2]
```diff
@@ -18,24 +18,52 @@ package main
...
-		hertzlogrus.WithTraceHookErrorSpanLevel(logrus.WarnLevel),
-		hertzlogrus.WithTraceHookLevels(logrus.AllLevels),
-		hertzlogrus.WithRecordStackTraceInSpan(true),
-	)
+	// Customizable output directory.
+	var logFilePath string
+	dir := "./hlog"
+	logFilePath = dir + "/logs/"
+	if err := os.MkdirAll(logFilePath, 0o777); err != nil {
+		fmt.Println(err.Error())
```
Comment (welkeyever) [inline hlog/logrus/main.go:42]: May return here? If the error happens here, there is no meaning for the rest of the DEMO?

Comment (li-jin-gou) [reply hlog/logrus/main.go:42]: Addendum: It is not recommended to use `fmt.Println` to print errors, we should use logging.

Comment (L2ncE) [reply hlog/logrus/main.go]: done

File: go.mod (hunk 1 above)
Comment (li-jin-gou) [reply go.mod:47]: add licenses

Comment (L2ncE) [reply go.mod:47]: > add licenses

done

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**